### PR TITLE
Anya/5055 datadog metrics

### LINKED ISSUE #5055: Ensure we can monitor attorney chekout launch
Let's ensure we can effectively monitor the health of the attorney phase 2 launch! We'll be deprecating Grafana, so let's not add any more monitoring there.

## AC 

- A Queue Business Metrics dashboard should be created with the following:
  - Attorney Case reviews completed (segment by OMO Request / Draft Decision) 
  - Judge case assignments to attorneys completed
- Ensure that we can monitor response time/response codes(error, success, etc) for the following routes in either New Relic or Datadog: 
  - GET /queue 
  - GET /queue/:user_id
  - GET /queue/tasks/:task_id
  - POST /tasks/:task_id/complete (we're interested in this because we think it might be very slow)

### PR DESCRIPTION
**Pull request by aroltsch**
Resolves #5055 

### Description
Queue Business Metrics

### Testing Plan
1. Installed Datadog locally
2. Collected metrics on all scenarios mentioned in the ticket
3. Checked to make sure updates made it to Datadog (env: local)

Dashboard is here: https://app.datadoghq.com/dash/774582/anya-test?live=true&page=0&is_auto=false&from_ts=1523281298096&to_ts=1523454098096&tile_size=m

### CODE REVIEW

File: app/services/business_metrics.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class BusinessMetrics
```
Comment (aroltsch) [inline app/services/business_metrics.rb]: I know we are planning on using tableau at some point but for the time being...

File: app/services/external_api/efolder_service.rb [hunk 2]
```diff
@@ -108,7 +108,7 @@ def self.send_efolder_request(endpoint, user, headers = {}, method: :get)
     request.headers = headers
     MetricsService.record("eFolder GET request to #{url}",
                           service: :efolder,
-                          name: endpoint) do
+                          name: "/api/#{endpoint.split("/")[2]}/#{endpoint.split("/")[3]}") do
```
Comment (aroltsch) [inline app/services/external_api/efolder_service.rb]: Mark, fixing eFolder service to avoid collecting metrics for specific IDs :)

Comment (mdbenjam) [reply app/services/external_api/efolder_service.rb]: Thank you! 🙏

File: app/services/business_metrics.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  def self.record(service: nil, name: "unknown")
+    app_name = RequestStore[:application] || "other"
+
+    DataDogService.increment_counter(
+      metric_group: "business",
+      metric_name: "event",
+      app_name: app_name,
+      attrs: {
+        service: service,
+        endpoint: name
```
Comment (mdbenjam) [inline app/services/business_metrics.rb]: We probably shouldn't call this endpoint. Business metrics are tied to API requests.

File: app/models/attorney_case_reviews/attorney_case_review.rb [hunk 4]
```diff
@@ -63,8 +62,12 @@ class << self
     def complete!(params)
       transaction do
         record = create!(params)
-        record.reassign_case_to_judge_in_vacols!
-        record.update_issue_dispositions! if record.type == "DraftDecision"
+        MetricsService.record("VACOLS: reassign_case_to_judge #{record.task_id}",
+                              service: :vacols,
+                              name: record.type) do
```
[the remark this answers is not in the document]
Comment (aroltsch) [reply app/models/attorney_case_reviews/attorney_case_review.rb]: Datadog makes all names snake case:

image

File: app/repositories/queue_repository.rb [hunk 5]
```diff
@@ -67,29 +66,30 @@ def self.decass_complexity_rating(vacols_id)
...
+      fail Caseflow::Error::QueueRepositoryError, msg
+    end
 
-      # update location with the judge's slogid
-      VACOLS::Case.find(vacols_id).update_vacols_location!(judge_vacols_user_id)
-      true
+    # In attorney checkout, we are automatically selecting the judge who assigned the attorney the case.
+    # But we also have a drop down for the attorney to select a different judge if they are checking it out to someone else
+    if decass_record.deadusr != judge_vacols_user_id
+      BusinessMetrics.record(service: :queue, name: "reassign_case_to_different_judge")
```
[the remark this answers is not in the document]
Comment (mdbenjam) [reply app/repositories/queue_repository.rb]: I think separating it makes sense. This seems like a different type of metric, and so a different abstraction feels right.

[the remark this answers is not in the document]
Comment (sunil-sadasivan) [reply app/repositories/queue_repository.rb]: Agreed on separating out this abstraction.  We could rename MetricsService to maybe `ServiceMetrics` just to keep the naming patterns consistent.  

eg. measuring Business Metrics and Service Metrics makes sense to me.

**Pull request merged**